### PR TITLE
KEP-24: Update AppArmor feature gates to GA stage.

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
As part of https://github.com/kubernetes/enhancements/issues/24 We are graduating AppArmor to GA in v1.31 or Kubernetes.

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/enhancements/issues/24

#### Does this PR introduce a user-facing change?

```release-note
Graduate Kubernetes' support for AppArmor to GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by tallclair** [comment]
Thanks. Can you also update these comments? https://github.com/kubernetes/kubernetes/blob/2f0cf23645876d39a6554f0cf6578a851c0a4891/pkg/features/kube_features.go#L62-L68

**Comment by sftim** [comment]
Changelog suggestion
```diff
-Graduate AppArmor to GA.
+Graduated AppArmor support to GA.
```

[AppArmor](https://apparmor.net/) is already GA; it's Kubernetes' support for it that is graduating.

**Comment by msau42** [comment]
api changes lgtm, can a sig lead approve the feature after verifying all the graduation criteria has been met?

**Comment by tallclair** [comment]
FYI, I'm proposing an additional change to AppArmor functionality in https://github.com/kubernetes/kubernetes/issues/125393. IMO this should not prevent going GA in the same release, but it would be good to avoid making this change after it's GA.

File: pkg/features/kube_features.go
Comment (SergeyKanzhelev) [inline pkg/features/kube_features.go:966]: I think there are more use of those feature gates than just util_test.go. Can you please delete those as well?

**Comment by Vyom-Yadav** [comment]
Hello @vinayakankugoyal! Just a reminder, the `code freeze is starting 02:00 UTC Wednesday 10th July 2024 (about 4 weeks from now)`, and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for `1.31`, is it still planned for this release?

File: pkg/features/kube_features.go
Comment (vinayakankugoyal) [reply pkg/features/kube_features.go:966]: I don't think we should remove that logic in this release. We should probably clean it up in the next release (1.32) as a post GA thing. 

@tallclair - what do you think?

**Comment by vinayakankugoyal** [comment]
> Hello @vinayakankugoyal! Just a reminder, the `code freeze is starting 02:00 UTC Wednesday 10th July 2024 (about 4 weeks from now)`, and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.
> 
> As the PR is tagged for `1.31`, is it still planned for this release?

Hi @Vyom-Yadav! Yes this PR is still planned for 1.31.

File: pkg/features/kube_features.go
Comment (vinayakankugoyal) [reply pkg/features/kube_features.go:966]: Discussed offline and kubernetes recommends that we remove all references to these in the code base as we are setting LockToDefault to true. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md?plain=1

Comment (vinayakankugoyal) [reply pkg/features/kube_features.go:966]: Done.

Comment (vinayakankugoyal) [reply pkg/features/kube_features.go]: Done. Thanks!

File: pkg/features/kube_features.go
Comment (vinayakankugoyal) [reply pkg/features/kube_features.go]: Done. Thanks!

**Comment by liggitt** [comment]
lgtm, will let tim ack before adding approve

**Comment by SergeyKanzhelev** [comment]
@vinayakankugoyal please add description to or remove this: https://github.com/kubernetes/kubernetes/blob/bd88faee8bb3b4d3c619bdd4ec7f1591abfc2bc2/test/e2e/feature/feature.go#L32

**Comment by vinayakankugoyal** [comment]
> Also I believe @tallclair wants to postpone this on [#125257 (comment)](https://github.com/kubernetes/kubernetes/pull/125257#issuecomment-2155381884)
> 
> /hold

I opened https://github.com/kubernetes/kubernetes/pull/125776 to address that. Can you take a look at that PR?

**Comment by vinayakankugoyal** [comment]
@SergeyKanzhelev - https://github.com/kubernetes/kubernetes/issues/125393 is now fixed. Can we proceed with this PR?

File: test/e2e_node/apparmor_test.go
Comment (SergeyKanzhelev) [inline test/e2e_node/apparmor_test.go:52]: is it running in any test lane without any labels? I think this is generic enough to mark as NodeConformance, this way it will run in conformance test lane

Comment (vinayakankugoyal) [reply test/e2e_node/apparmor_test.go:52]: Can you elaborate a bit on what you want me to do here? I am new to the testing infrastructure 😄

Comment (vinayakankugoyal) [reply test/e2e_node/apparmor_test.go:52]: Done.

**Comment by AdminTurnedDevOps** [comment]
Hello,

This issue/PR hasn't been updated in a while, so I just wanted to check the status. If there's anything we can do, please let us know.

Reminder: Code freeze starts `02:00 UTC Wednesday 24th July 2024 / 19:00 PDT Tuesday 23rd July 2024` (about 1.5 weeks from now). Please make sure the PR has both `lgtm` and `approved` labels before the code freeze.

**Comment by liggitt** [comment]
> @SergeyKanzhelev - #125393 is now fixed. Can we proceed with this PR?

@tallclair @SergeyKanzhelev are we holding or proceeding with this for 1.31?

**Comment by tallclair** [comment]
/hold cancel
/lgtm

Sorry for this delay. The soft admission change already merged (https://github.com/kubernetes/kubernetes/pull/125776). I don't see any need to postpone this.

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
